### PR TITLE
add bazaarpay gateway

### PR DESCRIPTION
**Pull request by farshadth**

### CODE REVIEW

File: src/Azki/Azki.php [hunk 1]
```diff
@@ -210,6 +216,48 @@ protected function verifyPayment()
...
+                    'provider_id' => $meta['provider_id'],
+                    'reaseon'     => !empty($params['reason']) ? $params['regret'] : self::DEFAULT_REFUND_REASON,
+                ],
+                "headers" => [
+                    'Signature'  => $this->generateSignature($subUrl),
+                    'MerchantId' => $this->config->get('azki.merchant-id'),
+                ],
+            ]);
+
+            $response = json_decode($response->getBody()->getContents());
```
Comment (m-jch) [inline src/Azki/Azki.php:245]: Please add a new log with information about refund when it's successful too

File: src/BazaarPay/BazaarPay.php [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                    'checkout_token' => $transaction->ref_id,
+                ],
+                'headers' => [
+                    'Authorization' => 'Token ' . $this->config->get('bazaarpay.token'),
+                    'Content-Type'  => 'application/json',
+                ]
+            ]);
+
+            $statusCode = $response->getStatusCode();
+            $response = $response->getBody()->getContents();
```
Comment (m-jch) [inline src/BazaarPay/BazaarPay.php:190]: Please add a new log with information about refund when it's successful too

**Review by m-jch — CHANGES_REQUESTED** [review]

File: src/Azki/Azki.php (hunk 1 above)
Comment (farshadth) [reply src/Azki/Azki.php:245]: done

File: src/BazaarPay/BazaarPay.php (hunk 2 above)
Comment (farshadth) [reply src/BazaarPay/BazaarPay.php:190]: done

**Pull request merged**